tests/cases/compiler/spreadsAndContextualTupleTypes.ts

```diff
@@ -0,0 +1,35 @@
+// @strict: true
+// @noEmit: true
+
+declare function fx4<T extends [string, string, string, 'a' | 'b']>(x: T): T;
+
+const t3 = ['x', 'y', 'z'] as const;
+
+fx4(['x', 'y', 'z', 'a']);
+fx4([...t3, 'a']);
+
+// Repro from #52684
+
+const staticPath1Level = ["home"] as const;
+const staticPath2Level = ["home", "user"] as const;
+const staticPath3Level = ["home", "user", "downloads"] as const;
+
+const randomID = 'id' as string;
+
+declare function foo<const T>(path: T): T;
+
+const a1 = foo([...staticPath1Level, randomID, 'doc.pdf']);
+const a2 = foo([...staticPath2Level, randomID, 'doc.pdf']);
+const a3 = foo([...staticPath3Level, randomID, 'doc.pdf']);
+
+const b1 = foo([...staticPath1Level, randomID, 'folder', 'doc.pdf']);
+const b2 = foo([...staticPath2Level, randomID, 'folder', 'doc.pdf']);
+const b3 = foo([...staticPath3Level, randomID, 'folder', 'doc.pdf']);
+
+const c1 = foo([...staticPath1Level, randomID, 'folder', 'subfolder', 'doc.pdf']);
+const c2 = foo([...staticPath2Level, randomID, 'folder', 'subfolder', 'doc.pdf']);
+const c3 = foo([...staticPath3Level, randomID, 'folder', 'subfolder', 'doc.pdf']);
+
+const d1 = foo([...staticPath1Level, randomID, 'folder', 'subfolder', 'another-subfolder', 'doc.pdf']);
+const d2 = foo([...staticPath2Level, randomID, 'folder', 'subfolder', 'another-subfolder', 'doc.pdf']);
+const d3 = foo([...staticPath3Level, randomID, 'folder', 'subfolder', 'another-subfolder', 'doc.pdf']);

```

Comment:
I thought that this PR might also fix [this issue](https://github.com/microsoft/TypeScript/issues/52760) but it doesn't. Perhaps it's something to work looking into here? The problem is similar but it's kinda a reverse one. This PR fixes contextual types for post-spread elements at the expression/runtime level and this other case is about post-spread elements at the type level.

The problem is that `getTypeOfPropertyOfContextualType` is still used here without any modifications as the primary source for the contextual type. The logic was only adjusted in the "else" branch here. The contextual type for this case is successfully retrieved through [those lines](https://github.dev/microsoft/TypeScript/blob/d8e81bbf6e6614bab7a0c5fba1e88a173943d22d/src/compiler/checker.ts#L28844-L28847) and since this doesn't take the position in the array expression into account - it's just `string`.

```suggestion
const d3 = foo([...staticPath3Level, randomID, 'folder', 'subfolder', 'another-subfolder', 'doc.pdf']);

// repro from #52760
const e1: [...string[], '!'] = ['a', '!'];

```

Reply:
This would require us to reason about elements at a fixed offset from the end of a tuple. We currently only reason about elements at a fixed offset from the start. It adds some more complexity, but it is certainly possible to do.